matter: pull Spake2pVerifier implementation for PSA (#19113)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

Matter
------

* Added implementation of the :cpp:class:`Spake2pVerifier` class for the PSA crypto backend.

>   This class can be used to generate the Spake2+ verifier at runtime.

Reviewer: ```suggestion
  You can use this class to generate the Spake2+ verifier at runtime.
```

>   To use this class, you must enable the :kconfig:option:`CONFIG_PSA_WANT_ALG_PBKDF2_HMAC` and :kconfig:option:`CONFIG_PSA_WANT_KEY_TYPE_SPAKE2P_KEY_PAIR_DERIVE` Kconfig options.

Reviewer: ```suggestion
  To use this class, enable the :kconfig:option:`CONFIG_PSA_WANT_ALG_PBKDF2_HMAC` and :kconfig:option:`CONFIG_PSA_WANT_KEY_TYPE_SPAKE2P_KEY_PAIR_DERIVE` Kconfig options.
```
